tests/lean/ppite.lean.expected.out

```diff
@@ -1,19 +1,13 @@
 def f : List Nat → IO Unit :=
 fun xs =>
-  List.forM xs
-    fun x =>
-      if (x == 0) = true then
-        do 
-          IO.println "foo"
-          IO.println "zero"
-      else
-        if (x % 2 == 0) = true then
-          do 
-            IO.println x 
-            IO.println "even"
-        else
-          do 
-            IO.println x 
-            IO.println "odd"
+  List.forM xs fun x =>
+    if (x == 0) = true then do
+      IO.println "foo"
+      IO.println "zero" else
+      if (x % 2 == 0) = true then do
+        IO.println x
+        IO.println "even" else do
+        IO.println x
+        IO.println "odd"
```

Comment:
Awesome!

Reply:
...oh, except for the `else`s? :)

Reply:
Okay, I've pushed a changed that puts newlines after then and before/after else (by exposing Format.group as a pretty-printing combinator, which requires a stage0 update).

There's still the structure instance issue, where the formatter produces `{ a := 42 b:= 1 }`.  I've tried replacing `optional ", "` by `", " <|> (ppLine >> group skip)`.  This produces parseable syntax, but formats every structure instance in multi-line format...  Ideally we'd define structure instances in terms of `sepBy ", " structInstField (", " <|> (checkColGe >> ppLine))` (so that the final `ppLine` is omitted), but that requires big changes in the structure instance elaborator (with the accompanying bootstrapping problems).